__tests__/app.test.js

```diff
@@ -99,68 +99,296 @@ describe("/api/articles/:article_id", () => {
+        favourite_colour: "blue",
+        votes: 5,
+      })
+      .expect(201)
+      .then(({ body }) => {
+        console.log(body.rows);
+        expect(body.rows[0].hasOwnProperty("article_id"));
+        expect(body.rows[0].hasOwnProperty("author"));
+        expect(body.rows[0].hasOwnProperty("body"));
+        expect(body.rows[0].hasOwnProperty("comment_id"));
+        expect(body.rows[0].hasOwnProperty("created_at"));
+        expect(body.rows[0].hasOwnProperty("votes"));
+        expect(!body.rows[0].hasOwnProperty("favourite_colour"));
+      });
+  });
+});
+
+describe("PATCH /api/articles/:article_id", () => {
+  test("PATCH 200: responds with a complete article object with all the correct properties", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ inc_votes: 5 })
+      .expect(200)
+      .then(({ body }) => {
+        const article = body[0];
+        expect(article.hasOwnProperty("article_id"));
+        expect(article.hasOwnProperty("title"));
+        expect(article.hasOwnProperty("topic"));
+        expect(article.hasOwnProperty("author"));
+        expect(article.hasOwnProperty("body"));
+        expect(article.hasOwnProperty("created_at"));
+        expect(article.hasOwnProperty("votes"));
+        expect(article.hasOwnProperty("article_img_url"));
+      });
+  });
+  test("PATCH 200: responds with an article object where the votes are increased by the correct amount", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ inc_votes: 5 })
+      .expect(200)
+      .then(({ body }) => {
+        const article = body[0];
+        expect(article.votes).toBe(105);
+      });
+  });
+  test("PATCH 200: responds with an article object where the votes are decreased by the correct amount", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ inc_votes: -10 })
+      .expect(200)
+      .then(({ body }) => {
+        const article = body[0];
+        expect(article.votes).toBe(90);
+      });
+  });
+  test("PATCH 400: responds with correct error when inc_vote is of invalid type", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({ inc_votes: "not a number" })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+  test("PATCH 400: responds with a correct error when article_id is not valid", () => {
+    return request(app)
+      .patch("/api/articles/numberone")
+      .send({ inc_votes: 5 })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+  test("PATCH 400: responds with correct error when inc_votes is not included in the request", () => {
+    return request(app)
+      .patch("/api/articles/1")
+      .send({})
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+});
```

Comment:
404 test needed for valid but non existing article_id :)

Reply:
on it chief